tasktracking/daml/ToDoList.daml

```diff
@@ -0,0 +1,325 @@
+daml 1.2
+module ToDoList where
+
+import DA.Time
+
+template ToDoItem
+    with 
+        issuer : Party
+        assignee : Party
+        toDoDescription : Text
+    where
+        signatory issuer, assignee
+
+        controller assignee can
+            StartWorkingOnAToDo : ContractId ToDoItemStartWorking
+                do 
+                    now <- getTime
+                    create ToDoItemStartWorking 
+                        with
+                            toDo = this
+                            startTime = now
+
+template ToDoItemProposal 
+    with 
+        toDo : ToDoItem
+    where 
+        signatory toDo.issuer
+
+        controller toDo.assignee can 
+            -- what happens if a task is not accepted? Do I need to reject it explicitly?
+            AcceptToDo : ContractId ToDoItem
+                do 
+                    create toDo 
+                        with 
+                            issuer = toDo.issuer
+                            assignee = toDo.assignee
+
+            RejectAndReturnToDo : ContractId ToDoItemProposal
+                -- by default assign the original issuer as the assignee letting him/her know that the proposal has been rejected
+                do
+                    create this
+                        with 
+                            toDo = this.toDo 
+                                with 
+                                    assignee = toDo.issuer
+            
+            TransferToDo : ContractId ToDoItemProposal
+                with newAssignee : Party
+                do
+                    create this
+                        with 
+                            toDo = this.toDo 
+                                with 
+                                    assignee = newAssignee
+
+template ToDoItemStartWorking
+    with 
+        toDo : ToDoItem
+        startTime : Time
+    where 
+        signatory toDo.issuer, toDo.assignee
+
+        controller toDo.assignee can
+            FinishWorking : ContractId ToDoItemFinishWorkingProposal
+                with
+                    newComment : Text
+                do
+                    now <- getTime
+                    create ToDoItemFinishWorkingProposal
+                        with 
+                            toDo = this.toDo
+                            startTime = this.startTime
+                            finishTime = now
+                            spentTime = subTime now this.startTime
+                            comment = newComment
+
+template ToDoItemFinishWorking
+    with 
+        toDo :  ToDoItem
+        startTime : Time
+        finishTime : Time
+        spentTime : RelTime
+    where 
+        signatory toDo.issuer
```

Comment:
Why don't you let the assignee sign this?

Reply:
@bame-da I would really have to dig to remember why I did it the way I did. There was a reason 😂